File: PublicSequenceMemoizer/src/main/java/edu/columbia/stat/wood/pub/sequencememoizer/IntSequenceMemoizerParameters.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package edu.columbia.stat.wood.pub.sequencememoizer;

import edu.columbia.stat.wood.pub.sequencememoizer.util.IntDiscreteDistribution;
import edu.columbia.stat.wood.pub.sequencememoizer.util.IntUniformDiscreteDistribution;

/**
 * Extension for int based Sequence Memoizer.
 * @author nicholasbartlett
 */
public class IntSequenceMemoizerParameters extends SequenceMemoizerParameters{

    /**
     * Base Distribution.
     */
    public IntDiscreteDistribution baseDistribution;

    /**
     * Constructor allowing all parameters to be set.
     * @param baseDistribution
     * @param discounts
     * @param infiniteDiscount
     * @param depth
     * @param seed
     * @param maxNumberRestaurants
     * @param maxSequenceLength
     */
    public IntSequenceMemoizerParameters(IntDiscreteDistribution baseDistribution, double[] discounts, double infiniteDiscount, int depth, long seed, long maxNumberRestaurants, long maxSequenceLength, long maxCustomersInRestaurant) {
        super(discounts, infiniteDiscount, depth, seed, maxNumberRestaurants, maxSequenceLength,maxCustomersInRestaurant);
        this.baseDistribution = baseDistribution;
    }

    /**
     * Constructor allowing some parameters to be set.  The default base distribution
     * is uniform over the types [0,alphabetSize).
     * @param depth
     * @param maxNumberRestaurants
     * @param maxSequenceLength
     * @param alphabetSize
     */
    public IntSequenceMemoizerParameters(int depth, long maxNumberRestaurants, long maxSequenceLength, int alphabetSize){
        super(depth, maxNumberRestaurants, maxSequenceLength);
        baseDistribution = new IntUniformDiscreteDistribution(alphabetSize);
    }

    /**
     * Constructor allowing some parameters to be set.  The default base distribution
     * is uniform over the types [0,alphabetSize).
     * @param depth
     * @param alphabetSize
     */
    public IntSequenceMemoizerParameters(int depth, int alphabetSize){
        super(depth, Long.MAX_VALUE, Long.MAX_VALUE);
        baseDistribution = new IntUniformDiscreteDistribution(alphabetSize);
    }

    /**
     * Constructor allowing some parameters to be set.  The default base distribution
     * is uniform over the types [0,alphabetSize).
     * @param alphabetSize
     */
    public IntSequenceMemoizerParameters(int alphabetSize) {
        super();
        baseDistribution = new IntUniformDiscreteDistribution(alphabetSize);
    }
}

File: R_SequenceMemoizer/R/IntSequenceMemoizerParameters.R
#' Class IntSequenceMemoizerParameters
#' 
#' Extension for int based Sequence Memoizer.
#' 
#' @slot ismp a edu.columbia.stat.wood.pub.sequencememoizer.IntSequenceMemoizerParameters object
setClass ("IntSequenceMemoizerParameters",
		  representation = representation (
				  ismp="jobjRef"))

#' Constructor of IntSequenceMemoizerParameters
#' 
#' @param alphabetSize integer or missing
#' @param depth integer or missing
#' @param maxNumberRestaurants integer or missing
#' @param maxSequenceLength integer or missing
#' @param ... Not used
setGeneric ("IntSequenceMemoizerParameters", function (alphabetSize, depth, maxNumberRestaurants, maxSequenceLength, ...) standardGeneric ("IntSequenceMemoizerParameters"))

#' Constructor allowing some parameters to be set. The default base distribution is uniform over the types [0,alphabetSize).
#' 
#' @param alphabetSize integer
setMethod ("IntSequenceMemoizerParameters", signature(alphabetSize="integer", depth="missing", maxNumberRestaurants="missing", maxSequenceLength="missing"),
		function (alphabetSize, depth, maxNumberRestaurants, maxSequenceLength, ...) {
			ismp <- .jnew ("edu.columbia.stat.wood.pub.sequencememoizer.IntSequenceMemoizerParameters", alphabetSize)
			new ("IntSequenceMemoizerParameters", ismp=ismp)

		})

#' Constructor allowing some parameters to be set. The default base distribution is uniform over the types [0,alphabetSize).
#'
#' @param alphabetSize integer 
#' @param depth integer
setMethod ("IntSequenceMemoizerParameters", signature(alphabetSize="integer", depth="integer", maxNumberRestaurants="missing", maxSequenceLength="missing"),
		function (alphabetSize, depth, maxNumberRestaurants, maxSequenceLength, ...) {
			ismp <- .jnew ("edu.columbia.stat.wood.pub.sequencememoizer.IntSequenceMemoizerParameters", depth, alphabetSize)
			new ("IntSequenceMemoizerParameters", ismp=ismp)
		})

#' Constructor allowing some parameters to be set. The default base distribution is uniform over the types [0,alphabetSize).
#' 
#' @param alphabetSize integer
#' @param depth integer
#' @param maxNumberRestaurants integer
#' @param maxSequenceLength integer
setMethod ("IntSequenceMemoizerParameters", signature(alphabetSize="integer", depth="integer", maxNumberRestaurants="integer", maxSequenceLength="integer"),
		function (alphabetSize, depth, maxNumberRestaurants, maxSequenceLength, ...) {
			ismp <- .jnew ("edu.columbia.stat.wood.pub.sequencememoizer.IntSequenceMemoizerParameters", depth, .jlong(maxNumberRestaurants), .jlong(maxSequenceLength), alphabetSize)
			new ("IntSequenceMemoizerParameters", ismp=ismp)
		})
